pkg/db/migration.go

```diff
@@ -125,8 +126,13 @@ func tryNewWithSourceInstance(sourceName string, sourceInstance source.Driver, d
 		if err == nil {
 			return m, nil
 		}
-
-		logging.Default().WithError(err).Error("could not open DB: Trying again")
+		netError := &net.OpError{}
+		if !(errors.As(err, &netError) && netError.Op == "dial") {
+			return nil, err
+		}
```

Comment:
Extract this repeated logic to a common place.

Reply:
I moved the logic to the errors.go file in the db package. I haven't seen code like this within other errors.go files, but an errors file seems like an appropriate place for code checking for errors. Is this okay?

The function is exported since it could conceivably be used elsewhere.

Reply:
This file is the right place. As for exporting it, I would usually make it unexported until the first external use-case (which sometimes forces you to move it to an entirely different package anyhow). Not blocking anyways..